upgrade/helpers/tools.py

```diff
@@ -233,21 +237,31 @@ def create_setup_dict(setups_dict):
     :param dict setups_dict: Dictionary of all return value of
     setup_products_for_upgrade
     """
-    with open('product_setup', 'w') as pref:
-        json.dump(setups_dict, pref)
+    data = {}
+
+    if os.path.exists(setup_data_file):
```

Comment:
consider using `pathlib.Path` instances for these file interactions.

Reply:
Done !